src/key.rs

```diff
@@ -589,6 +627,25 @@ impl PublicKey {
             }
         }
     }
+
+    /// Returns the [`XOnlyPublicKey`] (and it's [`Parity`]) for this [`PublicKey`].
+    #[inline]
+    pub fn x_only_public_key<C: Signing>(&self, secp: &Secp256k1<C>) -> Result<(XOnlyPublicKey, Parity), Error> {
```

Comment:
`split_xpubkey_parity`?

Reply:
I think that name is confusing and hard to remember.